src/envoy/tcp/mixer/filter.cc

```diff
@@ -170,20 +208,20 @@ bool Filter::GetDestinationIpPort(std::string *str_ip, int *port) const {
   }
   return false;
 }
+
 bool Filter::GetDestinationUID(std::string *uid) const {
   if (filter_callbacks_->upstreamHost()) {
     return Utils::GetDestinationUID(
         *filter_callbacks_->upstreamHost()->metadata(), uid);
   }
   return false;
 }
+
 const ::google::protobuf::Map<std::string, ::google::protobuf::Struct>
     &Filter::GetDynamicFilterState() const {
-  return filter_callbacks_->connection()
-      .streamInfo()
-      .dynamicMetadata()
-      .filter_metadata();
+  return cached_filter_metadata_;
```

Comment:
It seems that we only cache MongoProxy metadata and assume its all field types are "list" types.  This is too specific, not general enough for Istio.   

This is really a hack to solve MongoProxy metadata problem.  Maybe Istio/proxy is not the place to solve this problem.   It should be solved in MongoProxy

If we have to solve this in Istio/proxy,  it should NOT break other users that want to log other metadata.